core/src/main/java/com/linecorp/armeria/internal/AbstractHttp2ConnectionHandler.java

```diff
@@ -80,22 +79,21 @@ protected void onConnectionError(ChannelHandlerContext ctx, boolean outbound,
         }
 
         handlingConnectionError = true;
+        super.onConnectionError(ctx, outbound, cause, filterHttp2Exception(cause, http2Ex));
+    }
 
-        if (Flags.verboseResponses()) {
-            // TODO(trustin): Remove this once Http2ConnectionHandler.goAway() sends better debugData.
-            //                See https://github.com/netty/netty/issues/5160
-            if (http2Ex == null) {
-                http2Ex = new Http2Exception(INTERNAL_ERROR, goAwayDebugData(null, cause), cause);
-            } else if (http2Ex instanceof ClosedStreamCreationException) {
-                final ClosedStreamCreationException e = (ClosedStreamCreationException) http2Ex;
-                http2Ex = new ClosedStreamCreationException(e.error(), goAwayDebugData(e, cause), cause);
-            } else {
-                http2Ex = new Http2Exception(
-                        http2Ex.error(), goAwayDebugData(http2Ex, cause), cause, http2Ex.shutdownHint());
-            }
+    private static Http2Exception filterHttp2Exception(Throwable cause, @Nullable Http2Exception http2Ex) {
+        if (http2Ex != null) {
+            return http2Ex;
```

Comment:
`Http2Exception` and its subtypes already have enough debug data, so I skipped filtering.